packages/app/src/SearchPage.tsx

```diff
@@ -460,7 +460,9 @@ function SearchPage() {
     [searchInput],
   );
 
-  const [showSaveSearchModal, setShowSaveSearchModal] = useState(false);
+  const [saveSearchModalMode, setSaveSearchModalMode] = useState<
+    'update' | 'save' | 'hidden'
+  >('save');
```

Comment:
This need to be initialized as hidden or else it will show up every time the user opens the page